src/core/providers/strategies/helpers/signTransactions/helpers/guardTransactions/getCrossWindowProvider.ts

```diff
@@ -0,0 +1,25 @@
+import { CrossWindowProvider } from 'lib/sdkWebWalletCrossWindowProvider';
+
+export async function getCrossWindowProvider({
+  address,
+  walletUrl
+}: {
+  address: string;
+  walletUrl: string;
+}) {
+  try {
+    const success = await CrossWindowProvider.getInstance().init();
+    const provider = CrossWindowProvider.getInstance()
```

Comment:
Assign the provider instance into a variable. CrossWindowProvider.getInstance() called twice